csi.proto

```diff
@@ -264,6 +269,18 @@ message VolumeMetadata {
   // each Plugin keeps this information as small as possible.
   map<string, string> values = 1;
 }
+
+// A standard way to encode credential data. The total bytes of the values in
+// the Data field must be less than MaxSecretSize bytes.
+message Credentials {
+  // Data contains the credential data, for example username and password. Each
+  // key must consist of alphanumeric characters, '-', '_' or '.'. The
+  // serialized form of the secret data is a base64 encoded string, representing
+  // the arbitrary (possibly non-string) data value here.
+  // This information is sensitive and should be treated as such (not logged,
+  // etc.)
+  map<string, bytes> Data = 1;
```

Comment:
is it viable that an SP would advertise (via documentation) the credentials that it expects here? and then it's up to the CO (via operator or end-user specification) to include the appropriate details? if so then we should probably document the expectation more clearly

why the need for a standard enum?